primary/tests/epoch_change.rs

```diff
@@ -421,3 +421,115 @@ async fn test_restart_with_new_committee_change() {
         join_all(handles).await;
     }
 }
+
+/// Update the committee without changing the epoch.
+#[tokio::test]
+async fn test_simple_committee_update() {
+    let parameters = Parameters {
+        batch_size: 200, // Two transactions.
+        ..Parameters::default()
+    };
+
+    // The configuration of epoch 0.
+    let keys_0 = keys(None);
+    let committee_0 = pure_committee_from_keys(&keys_0);
+
+    // Spawn the committee of epoch 0.
+    let mut rx_channels = Vec::new();
+    let mut tx_channels = Vec::new();
+    for keypair in keys_0 {
+        let name = keypair.public().clone();
+        let signer = keypair;
+
+        let (tx_new_certificates, rx_new_certificates) = channel(CHANNEL_CAPACITY);
+        rx_channels.push(rx_new_certificates);
+        let (tx_feedback, rx_feedback) = channel(CHANNEL_CAPACITY);
+        tx_channels.push(tx_feedback.clone());
+
+        let initial_committee = ReconfigureNotification::NewCommittee(committee_0.clone());
+        let (tx_reconfigure, _rx_reconfigure) = watch::channel(initial_committee);
+
+        let store = NodeStorage::reopen(temp_dir());
+
+        Primary::spawn(
+            name,
+            signer,
+            Arc::new(ArcSwap::from_pointee(committee_0.clone())),
+            parameters.clone(),
+            store.header_store.clone(),
+            store.certificate_store.clone(),
+            store.payload_store.clone(),
+            /* tx_consensus */ tx_new_certificates,
+            /* rx_consensus */ rx_feedback,
+            /* dag */ None,
+            NetworkModel::Asynchronous,
+            tx_reconfigure,
+            /* tx_committed_certificates */ tx_feedback,
+            &Registry::new(),
+        );
+    }
+
+    // Run for a while in epoch 0.
+    for rx in rx_channels.iter_mut() {
+        loop {
+            let certificate = rx.recv().await.unwrap();
+            assert_eq!(certificate.epoch(), 0);
+            if certificate.round() == 10 {
+                break;
+            }
+        }
+    }
+
+    // Update the committee
+    let mut old_committee = committee_0;
+    for _ in 1..=3 {
+        // Update the committee
+        let mut new_committee = old_committee.clone();
+
+        let mut total_stake = 0;
+        let threshold = new_committee.validity_threshold();
+        for (_, authority) in new_committee.authorities.iter_mut() {
+            if total_stake < threshold {
+                authority.primary.primary_to_primary = format!(
+                    "/ip4/127.0.0.1/tcp/{}/http",
+                    config::utils::get_available_port()
+                )
+                .parse()
+                .unwrap();
+
+                total_stake += authority.stake;
+            }
+        }
+
+        // Notify the old committee to change epoch.
```

Comment:
`Notify the old committee to change epoch.` --> we don't want to change epoch right? Just update the existing committee

Reply:
good catch, fixing